### PR TITLE
Updating Test resources and fixing sample tests.

### PR DESCRIPTION
**Pull request by hamshavathimunibyraiah**
Update the test-resources.json to add role "Storage Blob Data Contributor" assigned to Translator resource because we no longer allow key based auth for storage accounts. Allowed Managed Identity on the translator resource.
Adding an overloaded method for DocumentTranslateContent constructor.
Fixing sample tests to work in test pipelines.
Updated TranslateWithAADAuth test to run in Playback mode only.

### CODE REVIEW

File: sdk/translation/test-resources.json [hunk 1]
```diff
@@ -77,13 +84,31 @@
...
+                "type": "SystemAssigned"
+            },
+            "properties": {
+                "customSubDomainName": "[variables('uniqueSubDomainName')]",
+                "networkAcls": {
+                    "defaultAction": "Allow",
+                    "virtualNetworkRules": [],
+                    "ipRules": []
+                },
+                "publicNetworkAccess": "Enabled"
```
Comment (benbp) [inline sdk/translation/test-resources.json:97]: This is now required to be false for compliance/security reasons.

```suggestion
```

File: sdk/translation/Azure.AI.Translation.Text/tests/TranslationLiveTests.cs [hunk 2]
```diff
@@ -557,6 +557,7 @@ public async Task TranslateWithTokenOptions()
         }
 
         [RecordedTest]
+        [PlaybackOnly("Live tests involving secrets will be temporarily disabled.")]
```
[the remark this answers is not in the document]
Comment (hamshavathimunibyraiah) [reply sdk/translation/Azure.AI.Translation.Text/tests/TranslationLiveTests.cs:560]: This needs some investigation on how to run AAD tests in Live mode. I could not find examples in the repo.
So created this task for tracking: https://machinetranslation.visualstudio.com/MachineTranslation/_workitems/edit/164263

File: sdk/translation/test-resources.json
[the remark this answers is not in the document]
Comment (hamshavathimunibyraiah) [reply sdk/translation/test-resources.json:1]: DT or TT, it for translator services. So I think it makes sense to have a single file.They both also reside in the same folder.
If you strongly feel they should be separate, lmk and I can create a task and work on it.

File: sdk/translation/test-resources.json [hunk 3]
```diff
@@ -115,7 +137,7 @@
             "sku": {
                 "name": "S1"
             },
-            "kind": "TextTranslation",
+            "kind": "TextTranslation",
```
[the remark this answers is not in the document]
Comment (hamshavathimunibyraiah) [reply sdk/translation/test-resources.json]: Fixed

**Review by MikeyMCZ — APPROVED** [review]

**Pull request merged**